### DISCUSSION TITLE
What is the correct way to change a body's position?

### DISCUSSION THREAD

**Discussion by captainkidd5** [amerkoleci/JoltPhysicsSharp#94] [Q&A] [answered]
I'm attempting to change the position of a sensor after it has been create by calling

BodyInterface.SetPosition(), but this deadlocks the program. If I use BodyInterfaceNoLock.SetPosition then I get an error that I'm trying to access protected memory. I'm looking at the single threaded access api and perhaps I need to use a lock somehow?

```
      Game1.PhysicsSim.PhysicsSystem.BodyLockInterface.LockWrite(_bigPlayerSensorBody.ID, out BodyLockWrite write);

      unsafe
      {
          void* body = write.Body.ToPointer();
          lock (body)
          {
              //?
          }
      }

```

**Comment by amerkoleci** [comment]
After creation? Doesn't Body.Position works?

**Comment by captainkidd5** [comment]
There's no set accessor on Body.Position directly, but it does work through the body interface, I was getting confused because 1). I forgot to add the body to the simulation (facepalm) and 2). The debug drawer does not appear to draw bodies when they are set to IsSensor = true.

**Reply by amerkoleci** [reply, marked as answer]
You don't need to write the body, just use BodyInterface.SetPosition 
https://github.com/amerkoleci/JoltPhysicsSharp/blob/main/src/JoltPhysicsSharp/BodyInterface.cs#L218

Jolt internally handles lock if I'm not wrong, try it
